File: 2019W06/preppindataw6.R
library(readxl)
library(dplyr)
library(lubridate)

new_sales <- read_excel("E:/Week6Input.xlsx", 
                        sheet = "England - Mar 2019") %>%
  mutate('Type of Soap'= sub('^(.*) Soap','\\1',`Category`),
         'Month'= format(as.Date("2019-03-01", "%Y-%m-%d"), format="%b %y"))  
price_dtl <- read_excel("E:/Week6Input.xlsx", 
                        sheet = "Soap Pricing Details") 
com_data <- read_excel("E:/Week6Input.xlsx", 
                       sheet = "Company Data") %>%
  mutate('Month'= format(`Month`, format="%b %d"))

final <- new_sales %>%
  merge(., price_dtl, by='Type of Soap') %>%
  group_by(`Month`,`Country`,`Category`) %>% 
  summarise('Profit'= sum(`Units Sold`*(`Selling Price per Unit`-`Manufacturing Cost per Unit`))) %>%
  ungroup %>%
  rbind(com_data,.)

View(final)

File: 2019W06/PreppindataW6.py
import pandas as pd

xls = pd.ExcelFile("E:\Week6Input.xlsx")
new_sales = pd.read_excel(xls,"England - Mar 2019") 
price_dtl = pd.read_excel(xls,"Soap Pricing Details") 
com_data = pd.read_excel(xls,"Company Data") 

temp = new_sales['Category'].str.split(' ', n = 1, expand = True) 
new_sales['Type of Soap'] = temp[0]

new_profit = new_sales.merge(price_dtl, on='Type of Soap', how='inner')
new_profit['Profit'] = new_profit['Units Sold']*(new_profit['Selling Price per Unit']-new_profit['Manufacturing Cost per Unit']) 
new_profit['Month'] = 'Mar 19'
new_profit = new_profit[['Month','Country','Category','Profit']].copy()
new_profit = new_profit.groupby(['Month','Country','Category'], as_index=False).agg({'Profit':'sum'})

com_data['Month']=com_data['Month'].dt.strftime('%b %y')
final = com_data.append(new_profit)
